### ISSUE TITLE
Check if libc hook callback is called from frida agent

### ISSUE THREAD

**Issue by 5z1punch** [frida/frida#946]
When I enable jit and hook some libc I/O function such as write, read, etc., the hooked app will crash quickly. 
Crash info in Logcat:

> 2019-07-05 12:50:20.557 1202-1204/? E/v8: #
    # Fatal error in , line 0
    # 
2019-07-05 12:50:20.557 1202-1204/? E/v8: Check failed: AllowJavascriptExecution::IsAllowed(isolate).
2019-07-05 12:50:20.557 1202-1204/? E/v8: #
    #
    #
    #FailureMessage Object: 0x764f21cac0
    
V8 Check failed caused crash. After difficult debugging I write a simple poc to show the key of the crash.

[poc for frida v8 crash](https://gist.github.com/5z1punch/444e23045fbf65b10169bd660c584c39)

The key of the problem is that when gc is called , V8 will disallow js execution to pause memory allocation request. But frida Interceptor.attach will trigger js callback when gc call `write` of libc. Then Check failed and crash.
In other words, we dont want to trigger our hook callback when hooked up function is called from frida (frida-agent.so) . 

I write a patch to solve the problem , But it only suit for Android. It check if the `*caller_ret_addr` is from the mem range of frida agent.

[init agent so mem address range](https://github.com/5z1punch/frida-core/commit/7259005f2795c8acb2ec1f00683c0b4a73e8cf11)
[check if hook callback is called from agent so](https://github.com/5z1punch/frida-gum/commit/1724918255ff1faa009a9f8160886a542a13523a)

I hope it will be help for someone else who has same problem. @oleavr
